docs/designs/repository-urls.md

```diff
@@ -0,0 +1,102 @@
+# Repository URLs
+
+Explain the 3 types of repository urls used in DocFX and their design purposes and use scenarios.   
+
+*******
+
+## Definitions
+
+| Name | Definition |
+|------|------------|
+| Repository Url | The url of current build repo, i.e. the argument of `git clone`  |
+| Publish Repository Url | The url of OPS provisioned repository |
+| Report Repository Url |  Specally used for reporting, it is the forked repo url for pull request build case |
+| Contribution Repository | Public repositories assoicated with the private provisioned repositories |
+
+	
+*******
+
+## Overview  
+The values of **Repository Url**, **Publish Repository Url** and **Report Repository Url**  will be passed via one or more the following approaches.
+1. DocFX command options
+2. stdin
+3. environment variables
+
+For all DocFX build scenarios, it will always fetch build information via ****Publish Repository Url****.
+
+**Build Outputs:** 
+ 1. DHS data   
+    Related DHS metadatas include _ContentGitUrl_, _OriginalContentGitUrl_ and _OriginalContentGitUrlTemplate_  which should be linked to the **Publish Repository Url** because their values will be calculated file content hash which is used in **incremental publish**.
+
+ 2. Report    
+    Including html report pages, report emails and PR comments.
+    Content authors may click links on report to navigate to the source files to fix content if there is any warning/error/suggestion.    
+    For PR builds, the links should point to the PR source repositiory url.    
+    The link urls will be generated from _.publish.json_ **source_url** properties which in turn are generated by prepending **Report Repository Url** for non-CRR cases.
+
+ 3. Telemetry    
```

Comment:
You can leave out the design on telemetry of Docs.Build from this document.

For DocFX, the telemetry code is at https://github.com/dotnet/docfx/blob/f20219e4b0c246a68ec3988dec4e7d75897cf2ab/src/docfx/lib/log/Telemetry.cs#L115-L115 where I think `Repository Url` should be good enough for now.